### ISSUE TITLE
Replacing my logo (image) with 'Django Administration'

### ISSUE THREAD

**Issue by cownaf** [geex-arts/django-jet#159]
Hi,

What is the best practice for replacing my company's logo file with the 'Django Administration' title?

Thx!

**Comment by SalahAdDin** [comment]
In the upper right corner?

**Comment by rcotrina94** [comment]
There are `site_header` and `site_title` properties of `Site` so I set an ADMIN_SITE_HEADER in settings, and this in the `urls.py` file:
```
if hasattr(settings, 'ADMIN_SITE_HEADER'):
    admin.site.site_header = settings.ADMIN_SITE_HEADER
    admin.site.site_title = settings.ADMIN_SITE_HEADER
```
This will change both, the visible title in the right top corner and the one in `<tittle>` of the site's HTML.

**Comment by Ismael-VC** [comment]
@rcotrina94 could you give me an example of what to put in `ADMIN_SITE_HEADER`?

I have `ADMIN_SITE_HEADER = '<img src="logo_bg.png" alt="BE Group logo">'` but I can't see the logo, just the normal title.

`site_header` and `site_title` are both `<django.utils.functional.lazy.<locals>.__proxy__ at 0x1f3ceade0f0>
` I'm not sure what to put there.

**Comment by SalahAdDin** [comment]
@rcotrina94, your solution doesn't work for logo, i think.

**Comment by Ismael-VC** [comment]
@cownaf  the correct way is *documented* here:

* https://github.com/geex-arts/django-jet-demo/blob/1a71b6f6ff327f39a0873902988979eb7163cc3f/templates/admin/base_site.html#L6-L10

You can create a `your_app/templates/admin/base_site.html`:

![image](https://cloud.githubusercontent.com/assets/4594825/23356388/3b09a4e2-fc9f-11e6-84ba-879724398848.png)

Something like:

```python
{% extends "admin/base_site.html" %}
{% load static i18n %}

{% block extrahead %}<link rel="shortcut icon" type="image/png" href="{{MEDIA_URL}}your_logo.png"/>{% endblock %}

{% block title %}{{ title }} |  {% trans "Your Title" %}{% endblock %}

{% block branding %}
        <h1 id="site-name">
            <a href="{% url 'admin:index' %}">
                <img style="background-color: white" src="{{MEDIA_URL}}your_logo.png" alt="your company name" height="50%" width="50%">
                <br><br>
                </span> {% trans "Your Branding" %}
            </a>
        </h1>
{% endblock %}
```

This achieves 3 things.

* Setup favicon:

```python
{% block extrahead %}<link rel="shortcut icon" type="image/png" href="{{MEDIA_URL}}your_logo.png"/>{% endblock %}
```

* Setup browser tab label:

```python
{% block title %}{{ title }} |  {% trans "Your Title" %}{% endblock %}
```

* Setup branding:

```python
{% block branding %}
        <h1 id="site-name">
            <a href="{% url 'admin:index' %}">
                <img style="background-color: white" src="{{MEDIA_URL}}your_logo.png" alt="your company name" height="50%" width="50%">
                <br><br>
                </span> {% trans "Your Branding" %}
            </a>
        </h1>
{% endblock %}
```

It is in this block where you can add your logo with an `<img ...> ` tag:

![image](https://cloud.githubusercontent.com/assets/4594825/23356976/77c74d60-fca1-11e6-8961-8e7af9d6ea67.png)

**Comment by SalahAdDin** [comment]
@Ismael-VC, yes please! With images also.

**Comment by Ismael-VC** [comment]
I've found a problem recently, same code as above, but now the logo is out of place, I'm using the latest dev branch:

![image](https://cloud.githubusercontent.com/assets/4594825/24831351/84feed0e-1c5d-11e7-9d5f-68a944fcdf14.png)

**Comment by SalahAdDin** [comment]
I think in the dev branch developers didn't take in account all possible uses for these block of code, in concecuence they create global styles for this menu icon only, and not for all block.

Can you try to review this? I want review also my[ issue](https://github.com/geex-arts/django-jet/issues/200).

**Comment by Ismael-VC** [comment]
@SalahAdDin yes I feel the same similarly with the footer that said **Powered by Django-JET**. Since this block was not inside an extra block tag, in order to change it, one would have to change the template itself, instead of inheriting from it.

Now this footer has been deleted in dev, but it would be nice to bring it back and make it editable, so the users can decide what to do with this extra branding blocks be it edit it, remove it or leave as is.

The weird thing is that the logo and the code I used was still working correctly, even after the update of the pin for the side menu was introduced, but after the last `pip install -r git+git://github.com/geex-arts/django-jet@dev
 --upgrade` the image gets moved downwards and I can't figure out why.

**Comment by SalahAdDin** [comment]
@Ismael-VC Did you review the code inspecting it in the browser?

**Comment by Ismael-VC** [comment]
Yes I did, I have reverted back to django-jet master for now, I want to test more things, but when I tested dev, and got this error, If I inspected it, it would tell me that the logo place is right between the two icons, even when you use the mouse selector, but the logo is moved downward non the less, I'll take a screenshot later today.

**Comment by SalahAdDin** [comment]
@Ismael-VC I tested the new release version, effectively, the problem exist yet:
![seleccion_328](https://cloud.githubusercontent.com/assets/5034215/25564484/efeccd64-2dbc-11e7-9d72-c85c3a05ce3b.png)

**Comment by SalahAdDin** [comment]
I partially solved with this:
![seleccion_330](https://cloud.githubusercontent.com/assets/5034215/25565714/4da8b414-2dd5-11e7-9673-b419c0169432.png)

**Comment by Ismael-VC** [comment]
Please come to the django-jet Discord server so we can organize if you like:

  *   https://discord.gg/3yr5yeH

Welcome! 😄

**Comment by mnislam01** [comment]
If I want to customize [This](https://github.com/geex-arts/django-jet/blob/dev/jet/dashboard/templates/jet.dashboard/modules/model_list.html) template by inheriting in my app how can I do it?

I guess this template shows models of an app in a list. I want to group some models for my application. May be different list for different group like that.

**Comment by sagata1999** [comment]
@SalchiPapa well, I tried, but after that default Dashboard died, it just dissappeared.
